portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -3218,6 +3218,24 @@ else if (groupId > 0) {
 			}
 		}
 
+		languageId = (String)httpServletRequest.getAttribute(
```

Comment:
@kevhlee This change should be covered by test.

If the reason you add this change as it was a bug, then you should expose the issue in test and fix it this way.

If the reason you made this change just because I18nFilter has it and you want to centralize, then you should figure out why I18nFilter has but PortalImpl does not, should we have it or not?

Reply:
@tinatian, I added this change because it covers a bug, specifically [LPS-175783](https://liferay.atlassian.net/browse/LPS-175783). The bug was addressed in `I18nFilter` but since we are centralizing, the part needs to moved. 

> This change should be covered by test.

Can you elaborate on this? I don't follow. I believe we are already covering this in `I18nFilterTest` with tests like `testGuestSpanishVirtualHostWithoutSessionCookieAlgorithm3`.

Reply:
@kevhlee So the reason you added it as it was a fix for a bug, you need to do:
1. Expose the bug in PortalImplTest as now the fix is in PortalImpl, or move the test from I18nFilterTest to PortalImplTest
2. Make this change and show that it is fixed